### PR TITLE
Add insert_all clause to match empty changesets list

### PR DESCRIPTION
**Pull request by rauann**
Currently `Oban.insert_all` does not handle when a wrapper is given with an empty list of changesets. This can be reproduced passing a wrapper like `Workflow` that has no changesets:
```elixir
** (FunctionClauseError) no function clause matching in Oban.insert_all/2    
    
    The following arguments were given to Oban.insert_all/2:
    
        # 1
        Oban
    
        # 2
        %Oban.Pro.Workers.Workflow{
          changesets: [],
          id: "2a464755-77dd-496c-ae68-d03ced6084bf",
          names: #MapSet<[]>,
          opts: %{}
        }
    
    Attempted function clauses (showing 2 out of 2):
    
        def insert_all(name, %{changesets: [_ | _] = changesets})
        def insert_all(name, changesets) when is_list(changesets)
    
    (oban 2.8.0) lib/oban.ex:302: Oban.insert_all/2
```

This PR adds new `insert_all/2` and `insert_all/4` functions clauses that handle empty changeset lists.

### CODE REVIEW

File: test/integration/inserting_test.exs [hunk 1]
```diff
@@ -89,4 +96,14 @@ defmodule Oban.Integration.InsertingTest do
...
   end
+
+  test "handling empty changesets list from wrapper using insert_all/4" do
+    name = start_supervised_oban!(queues: false)
+    wrap = %{changesets: []}
+
+    assert {:ok, %{}} =
+             name
+             |> Oban.insert_all(Ecto.Multi.new(), "jobs", wrap)
+             |> Repo.transaction()
```
Comment (sorentwo) [inline test/integration/inserting_test.exs:107]: No need to start a supervisor or run a transaction for this one either:

```suggestion
    multi = Ecto.Multi.new()

    assert ^multi = Oban.insert_all(Oban, multi, "jobs", %{changesets: []})
```

**Review by sorentwo — COMMENTED** [review]
Nice! People have asked about/requested this a few times. I've left a couple of comments on the tests. If those work, this is good to go!

File: lib/oban.ex [hunk 2]
```diff
@@ -308,6 +308,8 @@ defmodule Oban do
     insert_all(name, changesets)
   end
 
+  def insert_all(_name, %{changesets: []}), do: []
```
Comment (sorentwo) [inline lib/oban.ex:311]: Actually, looking at this in the full context I believe it's better to change the base `changesets` clause:

```elixir
  def insert_all(name, %{changesets: changesets}) when is_list(changesets) do
```

```elixir
  def insert_all(name, multi, multi_name, %{changesets: changesets}) when is_list(changesets) do
```

In which case, the tests should stay exactly as you wrote them 🙂

Comment (rauann) [reply lib/oban.ex]: Agreed, changed it :)

File: test/integration/inserting_test.exs (hunk 1 above)
Comment (sorentwo) [reply test/integration/inserting_test.exs:107]:
```suggestion
    name = start_supervised_oban!(queues: false)

    assert {:ok, %{jobs: []}} =
             name
             |> Oban.insert_all(Ecto.Multi.new(), :jobs, %{changesets: []})
             |> Repo.transaction()
```

File: test/integration/inserting_test.exs [hunk 3]
```diff
@@ -44,6 +44,13 @@ defmodule Oban.Integration.InsertingTest do
     [_job_1, _job_2] = Oban.insert_all(name, wrap)
   end
 
+  test "handling empty changesets list from wrapper with insert_all/2" do
+    name = start_supervised_oban!(queues: false)
+    wrap = %{changesets: []}
+
+    assert [] = Oban.insert_all(name, wrap)
```
[the remark this answers is not in the document]
Comment (sorentwo) [reply test/integration/inserting_test.exs:51]:
```suggestion
    name = start_supervised_oban!(queues: false)

    assert [] = Oban.insert_all(name, %{changesets: []})
```

**Review by sorentwo — APPROVED** [review]

**Comment by sorentwo** [comment]
Thanks @rauann 💛

**Pull request merged**